src/sip/dialog.c

```diff
@@ -475,14 +477,46 @@ int sip_dialog_update(struct sip_dialog *dlg, const struct sip_msg *msg)
 	if (sip_addr_decode(&addr, &contact->val))
 		return EBADMSG;
 
+	renc.mb = mbuf_alloc(512);
+	if (!renc.mb)
+		return ENOMEM;
+
 	err = pl_strdup(&uri, &addr.auri);
 	if (err)
 		return err;
 
-	if (dlg->route.scheme.p == dlg->uri) {
+	err = pl_strdup(&rtag, msg->req ? &msg->from.tag : &msg->to.tag);
+	if (err)
+		goto out;
+
+	renc.end = 0;
+
+	err |= sip_msg_hdr_apply(msg, msg->req, SIP_HDR_RECORD_ROUTE,
+				 record_route_handler, &renc) ? ENOMEM : 0;
```

Comment:
Thats looks a bit strange to me. If `record_route_handler` returns true, `ENOMEM` should returned?

Reply:
Absolutely, thanks. I will still clean this up a bit, hence the draft status. I wanted to make use of the GitHub workflows to make sure the test I wrote also works on other platforms.